File: trunk/esper/src/test/java/net/esper/support/filter/SupportFilterCallback.java
package net.esper.support.filter;

import net.esper.filter.FilterCallback;
import net.esper.event.EventBean;

public class SupportFilterCallback implements FilterCallback
{
    private int countInvoked;
    private EventBean lastEvent;

    public void matchFound(EventBean event)
    {
        countInvoked++;
        lastEvent = event;
    }

    public int getCountInvoked()
    {
        return countInvoked;
    }

    public EventBean getLastEvent()
    {
        return lastEvent;
    }

    public void setCountInvoked(int countInvoked)
    {
        this.countInvoked = countInvoked;
    }

    public void setLastEvent(EventBean lastEvent)
    {
        this.lastEvent = lastEvent;
    }

    public int getAndResetCountInvoked()
    {
        int count = countInvoked;
        countInvoked = 0;
        return count;
    }
}

File: NEsper.Tests/support/filter/SupportFilterCallback.cs
using System;

using net.esper.events;
using net.esper.filter;

namespace net.esper.support.filter
{
	
	public class SupportFilterCallback : FilterCallback
	{
		virtual public int CountInvoked
		{
			get
			{
				return countInvoked;
			}
			
			set
			{
				this.countInvoked = value;
			}
			
		}
		virtual public EventBean LastEvent
		{
			get
			{
				return lastEvent;
			}
			
			set
			{
				this.lastEvent = value;
			}
			
		}
		virtual public int AndResetCountInvoked
		{
			get
			{
				int count = countInvoked;
				countInvoked = 0;
				return count;
			}
			
		}
		private int countInvoked;
		private EventBean lastEvent;
		
		public virtual void  matchFound(EventBean _event)
		{
			countInvoked++;
			lastEvent = _event;
		}
	}
}
